src/mds/snap.h

```diff
@@ -103,11 +107,13 @@ struct sr_t {
   uint64_t change_attr = 0;             // tracks changes to snap
                                         // realm attrs.
 
-  __u32 flags = 0;
   enum {
     PARENT_GLOBAL	= 1 << 0,
     SUBVOLUME		= 1 << 1,
+    SNAPDIR_VISIBILITY = 1 << 2,
   };
+
+  __u32 flags = SNAPDIR_VISIBILITY; // snapdir visibility is always on by default
```

Comment:
Add a test case that:

* sets the snapdir visibility flag to false
*  fails the MDS
* confirm the flag is still set to false

Reply:
added, PTAL https://github.com/ceph/ceph/compare/a03ca82ead8b6128928769c45b10851821dfe7c2..c2047fc19e81d52eae0eb8130c7694572f6fbb21